src/types.rs

```diff
@@ -431,6 +502,186 @@ fn storage_key_value_hash_traces(key: U256, value: U256) -> [[Fr; 3]; 3] {
 }
 
 impl Proof {
+    pub fn old_account_leaf_hashes(&self) -> Option<Vec<Fr>> {
```

Comment:
is it needed to add more explanation here?  a lot of hardcoded index like old_account_hash_traces[4][0], old_account_hash_traces[0][2], new_account_hash_traces[2][1]

Reply:
yes. i was planning to add an account struct with methods like this:
```#[derive(Clone, Copy, Debug, Default)]
pub struct AccountData {
    balance: Fr, // We assume that all account balances can fit into 1 field element.
    keccak_codehash: U256,
    poseidon_codehash: Fr,
    code_size: u64,
    nonce: u64,
}

impl AccountData {
    fn hash(&self, storage_root: Fr) -> Fr {
        hash(
            self.poseidon_codehash_sibling(storage_root),
            self.poseidon_codehash,
        )
    }

    fn packed_nonce_and_codesize(&self) -> Fr {
        Fr::from(self.nonce) + Fr::from(self.code_size) * Fr::from(1 << 32).square()
    }

    fn hashed_nonce_codesize_balance(&self) -> Fr {
        hash(self.packed_nonce_and_codesize(), self.balance)
    }

    fn hashed_keccak_codehash(&self) -> Fr {
        let (high, low) = split_word(self.keccak_codehash);
        hash(high, low)
    }

    fn hashed_storage_root_keccak_codehash(&self, storage_root: Fr) -> Fr {
        hash(storage_root, self.hashed_keccak_codehash())
    }

    fn poseidon_codehash_sibling(&self, storage_root: Fr) -> Fr {
        hash(
            self.hashed_nonce_codesize_balance(),
            self.hashed_storage_root_keccak_codehash(storage_root),
        )
    }
}
```